### PR TITLE
compute-v1-firewall TF alternative

### LINKED ISSUE #653: Implement Terraform alternative for compute-v1-firewall snippets
Snippet to enhance: https://github.com/GoogleCloudPlatform/deploymentmanager-samples/blob/master/google/resource-snippets/compute-v1/firewall.yaml
More details: go/dm-snippets-alternatives
Example: #593, #627

### PR DESCRIPTION
**Pull request by egilliam**
TF alternative for compute-v1 firewall, and test script that deploys via both DM and TF and compares the results.

Closes #653.

### CODE REVIEW

File: google/resource-snippets/compute-v1/test_alternatives_firewall.sh [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+    diff /tmp/dm.yaml /tmp/tf.yaml
+    exit 1
+else
+    echo -e "${GREEN_COLOR}TF and DM outputs are identical${RESET_COLOR}"
+fi
+
+# TODO [#652]: Compare KRM and DM outputs.
+
+echo -e "${GREEN_COLOR}Test Success${RESET_COLOR}"
+exit 0
```
Comment (Doooritooos) [inline google/resource-snippets/compute-v1/test_alternatives_firewall.sh:69]: Please add a new line at the end of the file.

File: google/resource-snippets/compute-v1/alternatives-firewall/tf/firewall.tf [hunk 2]
```diff
@@ -0,0 +1,18 @@
+variable "deployment" {}
+variable "project_id" {}
```
Comment (Doooritooos) [inline google/resource-snippets/compute-v1/alternatives-firewall/tf/firewall.tf:2]: Please add the type and description to the two variables, see example below:
https://github.com/GoogleCloudPlatform/deploymentmanager-samples/blob/master/google/resource-snippets/pubsub-v1/alternatives/tf/pubsub.tf#L7-L17

File: google/resource-snippets/compute-v1/test_alternatives_firewall.sh (hunk 1 above)
Comment (egilliam) [reply google/resource-snippets/compute-v1/test_alternatives_firewall.sh]: Done.

File: google/resource-snippets/compute-v1/alternatives-firewall/tf/firewall.tf (hunk 2 above)
Comment (egilliam) [reply google/resource-snippets/compute-v1/alternatives-firewall/tf/firewall.tf]: Done.

**Review by Doooritooos — APPROVED** [review]

**Pull request merged**